drivers/SmartThings/matter-sensor/src/init.lua

```diff
@@ -157,8 +280,13 @@ local function do_configure(driver, device)
     profile_name = profile_name .. "-leak"
   end
 
-  if supports_battery_percentage_remaining(device) then
-    profile_name = profile_name .. "-battery"
+  local battery_feature_eps = device:get_endpoints(clusters.PowerSource.ID, {feature_bitmap = clusters.PowerSource.types.PowerSourceFeature.BATTERY})
+  if #battery_feature_eps > 0 then
+    if device:get_field(SUPPORT_BATTERY_PERCENTAGE) then
```

Comment:
Same issue as mentioned above, give the order in which these lifecycle events occur, I wouldn't expect this field to have been populated at this point since we'd still need to rely on a read response from the device. Even if the read request is sent and received before this function is called via `doConfigure`, there is still no guarantee of this order of events which could lead to a race condition. Therefore, I would rely on distinguishing between `battery` vs `battery-level` support in the AttributeList handler itself.

Reply:
In 061a1d0fa55282f8b120ebad74cd6a78d865f366, I moved the read request to `do_configure` after `match_profile` runs in order to avoid the race condition. Do you agree with this approach?

Reply:
I would update the attribute list handler in this driver and other to match what you did in the matter-lock driver!